src/shared/rm-rf.c

```diff
@@ -223,8 +294,13 @@ int rm_rf_children(
                          * We need to remove the inner directory we were operating on. */
                         assert(dirname);
                         r = unlinkat_harder(dirfd(todos[n_todo-1].dir), dirname, AT_REMOVEDIR, flags);
-                        if (r < 0 && r != -ENOENT && ret == 0)
-                                ret = r;
+                        if (r < 0 && r != -ENOENT) {
+                                if (ret == 0)
+                                        ret = r;
+
+                                if (FLAGS_SET(flags, REMOVE_CHMOD_RESTORE))
+                                        (void) fchmodat(dirfd(todos[n_todo-1].dir), dirname, old_mode & 07777, 0);
```

Comment:
maybe debug log on error?

Reply:
Currently, the function propagate the first error, and no later errors are logged. And in the code path, at least one error is already triggered.
Also, `TodoEntry` only contains the last component of the path, hence we need to call `fd_get_path()` to log about the failure...
I'd like to keep the code as is, at least in this PR.